### ISSUE TITLE
BLE connection with Wemos or Android?

### ISSUE THREAD

**Issue by condor13** [atc1441/ATC_TLSR_Paper#37]
How could I achieve a connection with an Wemos ?! would like to convert this to a strange looking "smartwatch" (just receive text notifications on it)

**Comment by nethomas1968** [comment]
I don't think the Hanshow devices are WiFi, but they have Bluetooth capabilities. And, I don't think the Wemos devices have bluetooth. I suspect that means that will have to connect the two devices together using old fashioned wires. The Hanshow devices can be communicated with using a serial interface. The Wemos has a number of GPIO pins that, I presume, can be configured to work in a serial manner. That would be my starting position I guess. I don't have any code examples on how to use GPIO pins on the Wemos as a serial port though.

**Comment by condor13** [comment]
Yes.. I know Hanshow devices dont have wifi, but you make a good point.. I could wire a serial connection. I know how to wire it so serial and how to write code to send serial from arduino.. but were should i look in this firmware in order to know how should I talk to it via serial.

**Comment by nethomas1968** [comment]
Hi,
You could possibly reverse engineer the JavaScript code from the Web Serial page at https://atc1441.github.io/ATC_TLSR_Paper_UART_Flasher.html

You can view the page source, and it shows you all the code . That might be a reasonable starting point for converting that to C or something more Wemos appropriate.

Regards

Nick
